### PR TITLE
support `AnyOf` with a single type

### PR DESCRIPTION
**Pull request by xavdid-stripe**
Some polymorphic fields only support a single variant initially. In order to keep a consistent interface for this field under the expectation that new values will be added, we need to support `AnyOf` expressions with only a single inner type.

### Method

I mostly copied `AnyOf<T1, T2>`, removing references to `T2` and changing `T1` to `T`. I also copied the tests, dropping some of the inputs and keeping the same assertions.

See: https://go/j/DEVSDK-2291

### CODE REVIEW

File: src/Stripe.net/Services/_base/AnyOf.cs
Comment (jar-stripe) [inline src/Stripe.net/Services/_base/AnyOf.cs:25]: It might make this type a bit more readable in the context of the file if we kept this as T1, with the idea being that we set the expectation that for any AnyOf, the first type is T1, etc.  Not a blocking thing, though.

**Review by jar-stripe — APPROVED** [review]

**Comment by xavdid-stripe** [comment]
@jar-stripe Are those test failures expected? I branched off latest `master` but I'm getting failures that seem related to your `ListOptions` changes?

File: src/Stripe.net/Services/_base/AnyOf.cs
Comment (xavdid-stripe) [reply src/Stripe.net/Services/_base/AnyOf.cs:25]: That's fair. I think in the context of the internals of this class, `T` makes more sense because there'll never be a `T2` to account for? But I don't feel super strongly either way.

**Comment by jar-stripe** [comment]
> @jar-stripe Are those test failures expected? I branched off latest `master` but I'm getting failures that seem related to your `ListOptions` changes?

Errors about ListOptions are expected (we haven't done a release since I merged those, right?)  Are there any other errors ?

**Comment by xavdid-stripe** [comment]
I don't think we've done a release, no. but it's fixed in master, right? so i'd expect PRs in the dotnet repo to have passing tests (but let me know if i'm misunderstanding)

File: src/Stripe.net/Services/_base/AnyOf.cs
Comment (jar-stripe) [reply src/Stripe.net/Services/_base/AnyOf.cs:25]: Same; I think your rationale makes sense.  The only exception is _maybe_ the case where I'm wading thru several variants of AnyOf and consistency here might make that easier, but I suspect the fact that this specific case hasn't come up yet means this doesn't/won't get touched often.

**Comment by jar-stripe** [comment]
> I don't think we've done a release, no. but it's fixed in master, right? so i'd expect PRs in the dotnet repo to have passing tests (but let me know if i'm misunderstanding)

Nah, we require a release here.  dotnet's compat check uses the package validation feature built into .NET, which checks a newly built .nupkg against the last package published to nuget (see https://learn.microsoft.com/en-us/dotnet/fundamentals/apicompat/package-validation/baseline-version-validator).

**Comment by xavdid-stripe** [comment]
Ah, got it! makes sense. 

I'm going to hold off merging to double check that this work will still be needed, but will merge once I know it is.

**Comment by xavdid-stripe** [comment]
Ok, we don't need this! The API shaped to be more correct.

**Pull request closed without merging**
